yarn-project/noir-libs/noir-aztec/src/note/lifecycle.nr

```diff
@@ -50,7 +52,13 @@ fn destroy_note<Note, N>(
         let serialise = note_interface.serialise;
         let preimage = serialise(note);
         assert(notify_nullified_note(storage_slot, nullifier, preimage) == 0);
+        
+        // We also need the note commitment corresponding to the "nullifier"
+        let compute_note_hash = note_interface.compute_note_hash;
+        let note_hash = compute_note_hash(note);
```

Comment:
By this point, we'll have already called `compute_note_hash` when we called `get`*, so it would be inefficient to call it again here.

@LeilaWang perhaps the `Note` needs to contain a member `note_hash: Option<Field>`, so that it can be retrieved by functions such as this one?

Reply:
Maybe also `nullifier: Option<Field>` too?

Reply:
Yeah I was aware that we're recomputing the note commitment. Didn't find a way to retrieve it while computing the nullifier. One way could be to change the `compute_nullifier` interface to return the pair (`nullifier`, `commitment`).

Reply:
I've added a TODO which we'll take care of as a separate PR.